### ISSUE TITLE
Platform context keys doesn't work for command removals

### ISSUE THREAD

**Issue by GandalfSaxe** [Microsoft/vscode#56382]
Issue Type: <b>Bug</b>

The new platform specific context key that words for key bindings (https://github.com/Microsoft/vscode/pull/54894/commits) doesn't work for "negative" commands, i.e. when the commands starts with a minus.

Easy way to reproduce:
1. Install [Markdown All In One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) extension.
2. Remap `markdown.etension.editing.toggleBold` to Ctrl+B (or something else)
3. Add `&& isMac` to end of "when" context of "positive command" - it still works! (⌘B toggles the sidebar)
4. Add `&& isMac` to end of "when" context of "negative command" - it doesn't work anymore! (⌘B toggles bold in Markdown)

In other words, this works:
```
{
    "key": "ctrl+b",
    "command": "markdown.extension.editing.toggleBold",
    "when": "editorTextFocus && !editorReadonly && editorLangId == 'markdown' && isMac"
  },
  {
    "key": "cmd+b",
    "command": "-markdown.extension.editing.toggleBold",
    "when": "editorTextFocus && !editorReadonly && editorLangId == 'markdown'"
  },
```

This doesn't work:

```
{
    "key": "ctrl+b",
    "command": "markdown.extension.editing.toggleBold",
    "when": "editorTextFocus && !editorReadonly && editorLangId == 'markdown' && isMac"
  },
  {
    "key": "cmd+b",
    "command": "-markdown.extension.editing.toggleBold",
    "when": "editorTextFocus && !editorReadonly && editorLangId == 'markdown' && isMac"
  },
```

VS Code version: Code - Insiders 1.27.0-insider (8727085fff33bf7d8f5e27483b10f72508d4167a, 2018-08-14T05:12:11.579Z)
OS version: Darwin x64 18.0.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz (8 x 2500)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: unavailable_software<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 4, 4|
|Memory (System)|16.00GB (0.15GB free)|
|Process Argv|/Users/gandalf/Documents/portable/Visual Studio Code - Insiders.app/Contents/MacOS/Electron|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
markdown-all-in-one|yzh|1.6.0

</details>

**Comment by GandalfSaxe** [comment]
Would suggest assignment to @Tyriar since he implemented #54894

**Comment by Tyriar** [comment]
@GandalfSaxe why are you using a when clause for this? Why not just:

```
  {
    "key": "cmd+b",
    "command": "-markdown.extension.editing.toggleBold"
  }
```

I suspect this is as designed since there's nothing special about isMac, it's just a regular context key (with a value that never changes).

**Comment by GandalfSaxe** [comment]
@Tyriar Basically I don't have a use case for it _right now_, but that's more coincidence than anything else.

The cmd key happen to not exist on Windows, so I can leave the command removal as you've quoted above without a when clause. But in principle one could be in a situation where one wanted to remove a key mapping from one platform, but not another. I don't see any reason why it shouldn't work though?

Actually now I have this:

```json
{
  // toggleItalic: KEY CONFLICT - overwrites expandLineSelection
  {
    "key": "ctrl+i",
    "command": "markdown.extension.editing.toggleItalic",
    "when": "editorTextFocus && !editorReadonly && editorLangId == 'markdown' && isMac"
  },
  {
    "key": "cmd+i",
    "command": "-markdown.extension.editing.toggleItalic",
    "when": "editorTextFocus && !editorReadonly && editorLangId == 'markdown'" // FIXME: Windows only when supported (https://github.com/Microsoft/vscode/issues/56382)
  },
  {
    "key": "ctrl+shift+alt+i",
    "command": "markdown.extension.editing.toggleItalic",
    "when": "editorTextFocus && !editorReadonly && editorLangId == 'markdown' && isWindows"
  },
  {
    "key": "ctrl+i",
    "command": "-markdown.extension.editing.toggleItalic",
    "when": "editorTextFocus && !editorReadonly && editorLangId == 'markdown'" // FIXME: Windows only when supported (https://github.com/Microsoft/vscode/issues/56382)
  }
}
```

The removal of ctrl+i for italic for Windows actually conflicts with settings of ctrl+i for macOS, but luckily the addition of ctrl+i takes higher precedence. Again feels like a stroke of luck, but it'd be more "correct" to have them be platform specific?

Anyway I don't have any _immediate_ use case for this feature, I just imagine it would be nice to have just in case - and I don't see any reason why it shouldn't work as one would expect for "negative commands". But if you're confident that there will never be a use case for this, you can of course ignore this 🙂

**Comment by Tyriar** [comment]
@GandalfSaxe I believe this is as designed as you cannot remove commands based on the when clause, they have to be the same. If you have keybinding `a` and `-b`, `a` still exists as they're different. It might seem unintuitive but that's the way the system was designed and it's fine imo once you understand this aspect.

Also on `cmd`, note that this will bind the super/windows key on windows and linux which may not be desirable. This is also by design I found as I created an issue on it recently, but luckily there's a workaround for that now.
